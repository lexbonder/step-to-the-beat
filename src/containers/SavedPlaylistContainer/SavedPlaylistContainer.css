.SavedPlaylistContainer button {
  display: block;
  margin: 10px;
}

.SavedPlaylistContainer .new-playlist {
  margin: 10px auto;
}

.no-playlists {
  color: #63B700;
  text-align: center;
  margin: 20px;
}