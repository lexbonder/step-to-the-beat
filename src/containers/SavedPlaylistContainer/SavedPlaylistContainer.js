import React, { Component } from 'react';
import SavedPlaylist from '../../components/SavedPlaylist/SavedPlaylist';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import './SavedPlaylistContainer.css';

export class SavedPlaylistContainer extends Component {
  
  savedPlaylistsToRender = () => {
    const { recentSeeds } = this.props;
    if (recentSeeds.length) {
      return recentSeeds.map(seed => <SavedPlaylist seed={seed} />);
    } else {
      return <h1>You do not have any favorited playlists!</h1>;
    }  
  }

  handleClick = () => {
    this.props.history.push('/select-spm');
  }

  render() {
    return (
      <div>
        <button onClick={this.handleClick}>
          Get New Playlist
        </button>
        <div>
          <h2>Nickname</h2>
          <h2>SPM</h2>
          <h2>Genre</h2>
          {this.savedPlaylistsToRender()}
        </div>
      </div>
    );
  }
}

export const MSTP = store => ({
  recentSeeds: store.recentSeeds
});

export default withRouter(connect(MSTP)(SavedPlaylistContainer));