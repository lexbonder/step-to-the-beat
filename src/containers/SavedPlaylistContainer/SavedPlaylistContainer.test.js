/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { SavedPlaylistContainer, MSTP } from './SavedPlaylistContainer';

describe('SavedPlaylistContainer', () => {
  let wrapper;
  let mockHistory = { push: jest.fn() }
  let mockRecentSeeds = [
    {spm: 148, genre: 'ska'},
    {spm: 160, genre: 'rock'}
  ]

  beforeEach(() => {
    wrapper = shallow(<SavedPlaylistContainer 
      history={mockHistory}
      recentSeeds={mockRecentSeeds}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('savedPlaylistsToRender', () => {
    it('should match the snapshot if there are recentSeeds', () => {
      expect(wrapper.instance().savedPlaylistsToRender()).toMatchSnapshot()
    })

    it('should match the snapshot if there are no recentSeeds', () => {
      const wrapper = shallow(<SavedPlaylistContainer 
        history={mockHistory}
        recentSeeds={[]}
      />)

      expect(wrapper.instance().savedPlaylistsToRender()).toMatchSnapshot()
    })
  })

  describe('handleClick', () => {
    it('should redirect the user to /select-spm', () => {
      wrapper.instance().handleClick()
      expect(mockHistory.push).toHaveBeenCalledWith('/select-spm')
    })
  })

  describe('MSTP', () => {
    it('should return an object with the values of whatever is in state', () => {
      const mockState = {recentSeeds: mockRecentSeeds}
      const mapped = MSTP(mockState)
      expect(mapped.recentSeeds).toEqual(mockRecentSeeds)
    })
  })
})