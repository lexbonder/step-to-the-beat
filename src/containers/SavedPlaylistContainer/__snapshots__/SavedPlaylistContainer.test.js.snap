// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedPlaylistContainer savedPlaylistsToRender should match the snapshot if there are no recentSeeds 1`] = `
<h1>
  You do not have any favorited playlists!
</h1>
`;

exports[`SavedPlaylistContainer savedPlaylistsToRender should match the snapshot if there are recentSeeds 1`] = `
Array [
  <withRouter(Connect(SavedPlaylist))
    seed={
        Object {
            "genre": "ska",
            "spm": 148,
          }
    }
/>,
  <withRouter(Connect(SavedPlaylist))
    seed={
        Object {
            "genre": "rock",
            "spm": 160,
          }
    }
/>,
]
`;

exports[`SavedPlaylistContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SavedPlaylistContainer
    history={
        Object {
            "push": [Function],
          }
    }
    recentSeeds={
        Array [
            Object {
              "genre": "ska",
              "spm": 148,
            },
            Object {
              "genre": "rock",
              "spm": 160,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
>
          Get New Playlist
</button>,
        <div>
          <h2>
                    Nickname
          </h2>
          <h2>
                    SPM
          </h2>
          <h2>
                    Genre
          </h2>
          <withRouter(Connect(SavedPlaylist))
                    seed={
                              Object {
                                        "genre": "ska",
                                        "spm": 148,
                                      }
                    }
          />
          <withRouter(Connect(SavedPlaylist))
                    seed={
                              Object {
                                        "genre": "rock",
                                        "spm": 160,
                                      }
                    }
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get New Playlist",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Get New Playlist",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Nickname
</h2>,
            <h2>
              SPM
</h2>,
            <h2>
              Genre
</h2>,
            Array [
              <withRouter(Connect(SavedPlaylist))
                seed={
                                Object {
                                                "genre": "ska",
                                                "spm": 148,
                                              }
                }
/>,
              <withRouter(Connect(SavedPlaylist))
                seed={
                                Object {
                                                "genre": "rock",
                                                "spm": 160,
                                              }
                }
/>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Nickname",
            },
            "ref": null,
            "rendered": "Nickname",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SPM",
            },
            "ref": null,
            "rendered": "SPM",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Genre",
            },
            "ref": null,
            "rendered": "Genre",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "seed": Object {
                "genre": "ska",
                "spm": 148,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "seed": Object {
                "genre": "rock",
                "spm": 160,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
>
            Get New Playlist
</button>,
          <div>
            <h2>
                        Nickname
            </h2>
            <h2>
                        SPM
            </h2>
            <h2>
                        Genre
            </h2>
            <withRouter(Connect(SavedPlaylist))
                        seed={
                                    Object {
                                                "genre": "ska",
                                                "spm": 148,
                                              }
                        }
            />
            <withRouter(Connect(SavedPlaylist))
                        seed={
                                    Object {
                                                "genre": "rock",
                                                "spm": 160,
                                              }
                        }
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get New Playlist",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Get New Playlist",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Nickname
</h2>,
              <h2>
                SPM
</h2>,
              <h2>
                Genre
</h2>,
              Array [
                <withRouter(Connect(SavedPlaylist))
                  seed={
                                    Object {
                                                      "genre": "ska",
                                                      "spm": 148,
                                                    }
                  }
/>,
                <withRouter(Connect(SavedPlaylist))
                  seed={
                                    Object {
                                                      "genre": "rock",
                                                      "spm": 160,
                                                    }
                  }
/>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Nickname",
              },
              "ref": null,
              "rendered": "Nickname",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SPM",
              },
              "ref": null,
              "rendered": "SPM",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Genre",
              },
              "ref": null,
              "rendered": "Genre",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "seed": Object {
                  "genre": "ska",
                  "spm": 148,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "seed": Object {
                  "genre": "rock",
                  "spm": 160,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
